src/aiida/tools/dumping/processes.py

```diff
@@ -0,0 +1,448 @@
+    """Simple helper function to generate the default parent-dumping directory if none given.
+
+    This function is not called for the sub-calls of `calcjob_dump` or during the recursive `workchain_dump` as it just
+    creates the default parent folder for the dumping, if no name is given.
+
+    :param process_node: The `ProcessNode` for which the directory is created.
+    :type process_node: WorkChainNode | CalcJobNode
+    :return: The created parent dump path.
+    :rtype: Path
+    """
+
+    try:
+        return Path(f'dump-{process_node.process_label}-{process_node.pk}')
+    except AttributeError:
+        # ? This case came up during testing, not sure how relevant it actually is
+        return Path(f'dump-{process_node.process_type}-{process_node.pk}')
+
+
+# ? Could move code from `generate_calcjob_node_io` here to normal function
+def attach_files_to_calcjob():
+    pass
+
+
+# ? Could move this to `cmdline/utils`
+def validate_make_dump_path(
+    path: Path = Path(),
+    overwrite: bool = False,
+) -> Path:
+    """
+    Create default dumping directory for a given process node and return it as absolute path.
+
+    :param path: The base path for the dump. Defaults to the current directory.
+    :type path: Path
+    :return: The created dump path.
+    :rtype: Path
+    """
+    import shutil
+
+    output_path = path.resolve()
+
+    # ? Use of `echo.echo_` only when running via `verdi`? -> I only see it used in the `cmd_` files.
+    if str(path) == '.' and overwrite:
+        echo.echo_critical('Path not set, defaults to CWD. Will not delete here for safety.')
+        return output_path
+
+    if output_path.is_dir() and any(output_path.iterdir()):
+        if overwrite:
+            # ? This might be a bit dangerous -> Check for it not being CWD enough?
+            # ? Added check for README in folder to decrease chances of deleting some other path
+            if (output_path / 'README').is_file():  # Also check for presence of README for safety
+                echo.echo_report(f'Overwrite set to true, will overwrite directory `{output_path}`.')
+                shutil.rmtree(output_path)
+            else:
+                echo.echo_critical(f'Something went wrong. Manually remove existing `{output_path}` and dump again.')
+                # echo.echo_critical(f'No README present in "{output_path}" Manually remove and dump again.')
+        else:
+            echo.echo_critical(f'Path `{output_path}` already exists and overwrite set to False.')
+
+    output_path.mkdir(parents=True, exist_ok=False)
+
+    return output_path
+
+
+def generate_node_input_label(index: int, link_triple: LinkTriple) -> str:
+    """Small helper function to generate the directory label for node inputs."""
+    node = link_triple.node
+    link_label = link_triple.link_label
+
+    # Generate directories with naming scheme akin to `verdi process status`
+    if link_label != 'CALL' and not link_label.startswith('iteration_'):
```

Comment:
Honestly, I don't even really remember why there is this check for 'CALL'. Could be that this was intended to check the `LinkType` instead, but even then it wouldn't make sense. I removed it. I left the check for `iteration_` as that would otherwise make the labels very bloated, and that should be a standard label dictated by `aiida-core` not by `aiida-quantumespresso` (at least I hope so ^^)?

Reply:
I guess the check for `CALL` is there because that is the default call link label, and so you will encounter this quite a bit. Call link labels do not have to be unique in AiiDA, so you can have a process with multiple outgoing call links with the label `CALL`.

This is also why I would advocate for removing this specific conditional because you might catch this case of `iteration_` but who knows what plugin packages do. They can choose any kind of call link label, even longer than that. In the end, I think it is consistency and predictability that is more important than in 1 particular case have slightly "nicer" directory names.

As it so happens, the `iteration_` prefix does not come from `aiida-quantumespresso` but it is used by the `BaseRestartWorkChain`:
https://github.com/aiidateam/aiida-core/blob/801715480a90e13fb60ade80b4d1002c163cd25e/src/aiida/engine/processes/workchains/restart.py#L207

But again, the call link label could be anything, so I would not add special behavior for just this single case.